header.go

```diff
@@ -216,12 +222,20 @@ func fixMangledMediaType(mtype, sep string) string {
 	}
 	parts := strings.Split(mtype, sep)
 	mtype = ""
-	for _, p := range parts {
-		if strings.Contains(p, "=") {
-			pair := strings.Split(p, "=")
-			if strings.Contains(mtype, pair[0]+"=") {
-				// Ignore repeated parameters.
-				continue
+	for i, p := range parts {
+		switch i {
+		case 0:
+			if p == "" {
+				// The content type is completely missing. Put in a placeholder.
+				p = ctPlaceholder
+			}
+		default:
```

Comment:
The real change is above this comment. The change below is just an indenting change to put it in the `default` case of the added `switch`.